sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingCompletions.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Sse;
+
+namespace Azure.AI.OpenAI.Custom
+{
+    public class StreamingCompletions : IDisposable
```

Comment:
Should we represent these models in the cadl?  If that is the plan for future lets double check the shape to ensure that when we add them they can be removed from custom code and generated

Reply:
Absolutely--the goal is to do as much of this as possible from cadl as soon as possible. I'll follow up on the chat thread(s) about double-checking the surface; for preview, the scales likely tilt towards getting streaming support in even if we break it in a bit, but that doesn't mean we shouldn't avoid low-hanging fruit splatting down on us if we can.

Reply:
Per quick response off-PR, this'll be an important consideration between now and exiting preview -- but not a blocker for this PR so much as an important note.